src/median.ts

```diff
@@ -0,0 +1,103 @@
+import type {
+  IterableContainer,
+  NonEmptyArray,
+  NonEmptyReadonlyArray,
+} from "./internal/types";
+import { isBigInt } from "./isBigInt";
+import { isNumber } from "./isNumber";
+import { mean } from "./mean";
+import { purry } from "./purry";
+import { sort } from "./sort";
+
+type Median<T extends IterableContainer<bigint> | IterableContainer<number>> =
+  T extends NonEmptyArray<bigint> | NonEmptyReadonlyArray<bigint>
+    ? bigint
+    : T extends NonEmptyArray<number> | NonEmptyReadonlyArray<number>
+      ? number
+      : T extends [] | readonly []
+        ? undefined
+        : number | undefined;
```

Comment:
**question:** actually, shouldn't this depend on whether `T` is an array of `bigint`s or `number`s? and same with `mean`?

Reply:
I think you are right, would you suggest something like the following: 
```
T extends [] | readonly []
    ? undefined
    : T[number] extends bigint
         ? bigint | undefined
         : number | undefined
```
and the related type tests for bigints should be modified as follows:
```
test("arbitrary bigint arrays", () => {
    const result = pipe([] as Array<bigint>, median());
    expectTypeOf(result).toEqualTypeOf<bigint | undefined>(); // instead of <number | undefined>
});
```

Reply:
i think so! and rmb to do the same with `mean`

Reply:
Thank you, should I open a new PR to integrate this change into `mean` ?